components/activity_and_insights/insights/top_reactions/top_reactions.tsx

```diff
@@ -1,47 +1,116 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
-import React, {memo} from 'react';
+import React, {memo, useEffect, useState, useCallback} from 'react';
+import {shallowEqual, useDispatch, useSelector} from 'react-redux';
+import {FormattedMessage} from 'react-intl';
+
+import Icon from '@mattermost/compass-components/foundations/icon/Icon';
+
+import {getTopReactionsForTeam, getMyTopReactions} from 'mattermost-redux/actions/insights';
+import {getTopReactionsForCurrentTeam, getMyTopReactionsForCurrentTeam} from 'mattermost-redux/selectors/entities/insights';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {TopReaction} from '@mattermost/types/insights';
+
+import {InsightsScopes} from 'utils/constants';
 
 import BarChartLoader from '../skeleton_loader/bar_chart_loader/bar_chart_loader';
 import CircleLoader from '../skeleton_loader/circle_loader/circle_loader';
-import widgetHoc from '../widget_hoc/widget_hoc';
+import widgetHoc, {WidgetHocProps} from '../widget_hoc/widget_hoc';
+
+import TopReactionsBarChart from './top_reactions_bar_chart/top_reactions_bar_chart';
 
 import './../../activity_and_insights.scss';
 
-// eslint-disable-next-line no-empty-pattern
-const TopReactions = ({}) => {
+const TopReactions = (props: WidgetHocProps) => {
+    const dispatch = useDispatch();
+
+    const [loading, setLoading] = useState(true);
+    const [topReactions, setTopReactions] = useState([] as TopReaction[]);
+
+    const teamTopReactions = useSelector((state: GlobalState) => getTopReactionsForCurrentTeam(state, props.timeFrame, 5), shallowEqual);
+    const myTopReactions = useSelector((state: GlobalState) => getMyTopReactionsForCurrentTeam(state, props.timeFrame, 5), shallowEqual);
```

Comment:
Is this inefficient to invoke both selectors during the component's render phase?

Reply:
Invoking useSelector every render has to be predictable, you can't wrap them in a condition unfortunately. I did try doing this:
```
const myTopReactions = useSelector((state: GlobalState) => {
        if (props.filterType === InsightsScopes.MY) {
            return getMyTopReactionsForCurrentTeam(state, props.timeFrame, 5);
        }
        return [];
    }, shallowEqual);
```
but it didn't make a difference to re-renders

Reply:
You could make a selector that also takes the `props.filterType` and returns whichever of the values you need. That way, you also wouldn't need the intermediate `topReactions` state